search/src/main/java/edu/artic/search/CircularViewItemDecoration.kt

```diff
@@ -0,0 +1,47 @@
+package edu.artic.search
+
+import android.graphics.Rect
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import com.fuzz.rx.DisposeBag
+import com.fuzz.rx.disposedBy
+
+class CircularViewItemDecoration(private val currentPadding: Int,
+                                 private val finalPadding: Int,
+                                 private val disposeBag: DisposeBag) : RecyclerView.ItemDecoration() {
+
+    private var onTheMapPosition = 0
+
+    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
+        val position = parent.getChildAdapterPosition(view)
+        val adapter = parent.adapter as SearchResultsAdapter
+
+        adapter.getItemOrNull(position)?.let {
+            if (it is SearchTextHeaderViewModel) {
+                it.text.subscribe {
+                    if (it == R.string.on_the_map) {
+                        onTheMapPosition = position
+                    }
+                }.disposedBy(disposeBag)
+            }
+            if (onTheMapPosition > 0) {
+                if (it is SearchCircularCellViewModel || it is SearchAmenitiesCellViewModel) {
```

Comment:
Move this `if` up a line, turn into an `else if` with the `SearchTextHeaderViewModel` block.